src/wallet/interfaces.cpp

```diff
@@ -154,10 +154,7 @@ class WalletImpl : public Wallet
     std::string getWalletName() override { return m_wallet->GetName(); }
     bool getNewDestination(const std::string label, CTxDestination& dest) override
     {
-        auto spk_man = m_wallet->GetLegacyScriptPubKeyMan();
-        if (!spk_man) {
-            return false;
-        }
+        LOCK(m_wallet->cs_wallet);
```

Comment:
`cs_wallet` is locked inside `GetNewDestination`, no need to lock it here
```suggestion
```

Reply:
I keep it for unification with bitcoin's codebase. Even now [in the master](https://github.com/bitcoin/bitcoin/blob/master/src/wallet/interfaces.cpp#L156-L157) they still have this `cs_wallet` locked.
```
    util::Result<CTxDestination> getNewDestination(const OutputType type, const std::string& label) override
    {
        LOCK(m_wallet->cs_wallet);
        return m_wallet->GetNewDestination(type, label);
    }
```

Reply:
oh, ok, nvm then